deeplake/core/vectorstore/vector_search/ingestion/data_ingestion.py

```diff
@@ -0,0 +1,144 @@
+from typing import Dict, List, Any, Callable, Optional
+
+import numpy as np
+
+import deeplake
+from deeplake.core.dataset import Dataset as DeepLakeDataset
+from deeplake.core.vectorstore.vector_search import utils
+from deeplake.util.exceptions import TransformError
+from deeplake.constants import MAX_RETRY_ATTEMPTS, MAX_CHECKPOINTING_INTERVAL
+
+
+class DataIngestion:
+    def __init__(
+        self,
+        elements: List[Dict[str, Any]],
+        dataset: DeepLakeDataset,
+        embedding_function: Optional[Callable],
```

Comment:
just a side note embedding function is needed for testing purposes, I am not planning to use it during integration with LangChain or LamaIndex, there the embedding function logic will be separated